lib/ansible/modules/packaging/language/pip.py

```diff
@@ -384,7 +384,7 @@ def main():
 
     venv_created = False
     if chdir:
-        env = os.path.join(chdir, env)
+        env = os.path.join(chdir, env or '')
```

Comment:
Do you think we should add additional check for `env` variable so that it doesn't silently takes blank value ?

Reply:
Imo having `type=path` is enough validation for `virtualenv` or what kind of blank values do you mean?

Reply:
I am not sure, this solves the problem as intended, because it might accidentially set env to some path (chdir) if it was supposed to be None (aka do not use / create a virtual env).

Reply:
As a reference, @mdellweg suggests https://github.com/ansible/ansible/pull/39595/files
I like the solution by @mdellweg more. @tonal what's your opinion?

Reply:
I think @mdellweg is right.
How to about both set `chdir` and `executable`?
It seems that in this case, too, you need to joined the paths:
```
    executable = module.params['executable']
    if chdir:
        if env:
            env = os.path.join(chdir, env)
        if executable:
            executable = os.path.join(chdir, executable)
### and after
    pip = _get_pip(module, env, executable)
```